File: php/testcode/securecookie_003.php
<?php
require_once __DIR__ . '/shared.php';

// vuln-code-snippet start php_cookie_session_default
function securecookie003(BenchmarkRequest $req): BenchmarkResponse {
    session_start(); // vuln-code-snippet vuln-line php_cookie_session_default
    $_SESSION['user'] = $req->param('user');
    return BenchmarkResponse::ok('Session started');
}
// vuln-code-snippet end php_cookie_session_default

File: ruby/testcode/securecookie_003.rb
require_relative 'shared'

# vuln-code-snippet start ruby_securecookie_no_httponly
def set_session(req)
  session_id = req.param('session_id')
  response = BenchmarkResponse.ok('session started')
  response.headers['Set-Cookie'] = "session=#{session_id}; Secure"  # vuln-code-snippet vuln-line ruby_securecookie_no_httponly
  response
end
# vuln-code-snippet end ruby_securecookie_no_httponly
